### PR TITLE
Upgrading all the deps

### LINKED ISSUE #166: Move to fuzzy dependency versions
The package.json pins down dependencies pretty hard. This leads to duplicate dependencies in our project, e.g. `prop-types` which already has newer releases. Would you consider to updating all dependencies with carets `^` or at least tildes `~`? Or is there any reason why pinning them down, if you rely on yarn anyway.

PS: I see that you have travis-ci in place, you could think about adding [greenkeeper](https://greenkeeper.io/) as well to update dependencies automatically and report breakages with newer versions of this package.

### PR DESCRIPTION
**Pull request by alexreardon**
As the name says - this PR upgrades all the dependencies to the latest version (except for react).

Previously this was blocked by #37 

- [x] All deps
- [x] eslint to latest
- [x] flow to latest
- [x] Move from npm ignore file to files array in package json (moving from black listing to white listing)
- [x] Removing (unused) prop type generation
- [x] Moving to fuzzy dependencies (closes #166)

I also updated the `ISSUE_TEMPLATE` as per our discussion @jaredcrowe

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1,37 +0,0 @@
-# Editors
```
Comment (alexreardon) [inline .npmignore]: Now moved to whitelisting in package.json

File: src/view/droppable-dimension-publisher/droppable-dimension-publisher.jsx [hunk 2]
```diff
@@ -47,10 +45,12 @@ export default class DroppableDimensionPublisher extends Component {
       isDropDisabled,
       targetRef,
     } = this.props;
-    invariant(targetRef, 'DimensionPublisher cannot calculate a dimension when not attached to the DOM');
```
Comment (jaredcrowe) [inline src/view/droppable-dimension-publisher/droppable-dimension-publisher.jsx]: why get rid of invariant here but not elsewhere in this file where it's used?

Comment (alexreardon) [reply src/view/droppable-dimension-publisher/droppable-dimension-publisher.jsx]: Good call - i will look at why this was done. I think i was battling a flow issue.

File: package.json [hunk 3]
```diff
@@ -15,6 +15,9 @@
     "url": "https://github.com/atlassian/react-beautiful-dnd/issues"
   },
   "main": "lib/index.js",
+  "files": [
+    "/lib"
```
Comment (jaredcrowe) [inline package.json]: publishing this now?

File: src/view/droppable-dimension-publisher/droppable-dimension-publisher.jsx (hunk 2 above)
Comment (alexreardon) [reply src/view/droppable-dimension-publisher/droppable-dimension-publisher.jsx]: I think it has to do with flow not understanding invariant

File: package.json (hunk 3 above)
Comment (alexreardon) [reply package.json]: yeah - just the lib 💃 
I did a test run and it looks good

Comment (jaredcrowe) [reply package.json]: oh this is because we got rid of the .npmignore?

File: src/view/draggable/draggable-types.js [hunk 4]
```diff
@@ -109,18 +109,18 @@ export type StateSnapshot = {|
 // Using PropType<*,*> to allow strict typing within code and looser typing
```
Comment (jaredcrowe) [inline src/view/draggable/draggable-types.js]: can probably remove this comment now?

File: package.json (hunk 3 above)
Comment (alexreardon) [reply package.json]: you can run `yarn pack` to see what will be published

File: src/view/draggable/draggable-types.js (hunk 4 above)
Comment (alexreardon) [reply src/view/draggable/draggable-types.js]: nice. done

File: package.json [hunk 5]
```diff
@@ -29,50 +32,45 @@
     "prepublish": "yarn run build"
   },
   "dependencies": {
-    "babel-plugin-react-flow-props-to-prop-types": "0.15.0",
-    "babel-runtime": "6.26.0",
-    "invariant": "2.2.2",
-    "memoize-one": "3.0.1",
-    "prop-types": "15.5.10",
```
Comment (jaredcrowe) [inline package.json]: I don't think we can remove prop-types, it's still used in some places... (typing context types for instance)

File: package.json [hunk 6]
```diff
@@ -29,50 +32,45 @@
...
-    "jest": "21.1.0",
-    "raf-stub": "2.0.1",
-    "react": "15.6.1",
-    "react-dom": "15.6.1",
-    "react-test-renderer": "15.6.1",
-    "rimraf": "2.6.2",
-    "styled-components": "2.1.2"
-  },
-  "peerDependencies": {
-    "react": ">=15.4.0"
```
Comment (jaredcrowe) [inline package.json]: keep react as a peerdep rather than a devdep?

Comment (alexreardon) [reply package.json]: Kept it as both - i think we needed it for storybook

File: package.json (hunk 5 above)
Comment (alexreardon) [reply package.json]: Correct. I have re-added it

File: package.json [hunk 7]
```diff
@@ -29,50 +32,46 @@
     "prepublish": "yarn run build"
   },
   "dependencies": {
-    "babel-plugin-react-flow-props-to-prop-types": "0.15.0",
```
Comment (jaredcrowe) [inline package.json]: we don't need this any more?

Comment (alexreardon) [reply package.json]: This is probably the most contentious part of the PR. This is used to generate prop types from flow. However, it is just one more thing to maintain and makes it a little hard to move forward quickly. My thought is to drop it given that we publish flow types and there is public typescript types also. 

What are your thoughts?

Comment (alexreardon) [reply package.json]: Update: we were not generating prop types anyway. Given that nobody has complained I think we just leave it as is

**Pull request merged**